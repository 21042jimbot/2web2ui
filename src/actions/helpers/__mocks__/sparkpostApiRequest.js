const mock = jest.fn((action) => action);

export default mock;
