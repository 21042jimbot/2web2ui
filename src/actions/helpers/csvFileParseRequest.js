import _ from 'lodash';
import Papa from 'papaparse';

export const hasData = ({ data }) => {
  if (data.length === 0) {
    return 'No data was found.';
  }
};

export const hasField = (...expectedFields) => ({ meta: { fields }}) => {
  if (!expectedFields.some((field) => fields.includes(field))) {
    return `Missing required field, ${expectedFields[0]}.`;
  }
};

export default ({ meta: { file, validate }, type, parser = Papa }) => async(dispatch) => {
  const types = {
    PENDING: `${type}_PENDING`,
    SUCCESS: `${type}_SUCCESS`,
    FAIL: `${type}_FAIL`
  };

  dispatch({ type: types.PENDING });

  return await new Promise((resolve, reject) => parser.parse(file, {
    complete: ({ data, errors, meta }) => {
      let error;

      if (errors.length > 0) {
        error = new Error('An error occurred while parsing your file.');

        dispatch({
          type: types.FAIL,
          payload: {
            message: error.message,
            details: errors
          }
        });

        reject(error);
        return;
      }

      // Return the first validation error message
      const message = _.find(validate, (v) => v({ data, meta }));

      if (message) {
        error = new Error(message);

        dispatch({ type: types.FAIL, payload: { message }});
        reject(error);

        return;
      }

      dispatch({ type: types.SUCCESS, payload: { data }});
      resolve(data);
    },
    error: (error) => {
      dispatch({ type: types.FAIL, payload: { message: error.message }});
      reject(error);
    },
    header: true,
    skipEmptyLines: true
  }));
};
