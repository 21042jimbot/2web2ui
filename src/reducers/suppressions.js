const initialState = { list: [], listLoading: true };

export default (state = initialState, action) => {
  switch (action.type) {
    case 'GET_SUPPRESSIONS_PENDING':
      return { ...state, listLoading: true, listError: null };

    case 'GET_SUPPRESSIONS_SUCCESS':
      return { ...state, list: action.payload, listLoading: false };

    case 'GET_SUPPRESSIONS_FAIL':
      return { ...state, listError: action.payload, listLoading: false };

    // search suppressions
    case 'SEARCH_SUPPRESSIONS_RECIPIENT_PENDING':
      return { ...state, loading: true };

    case 'SEARCH_SUPPRESSIONS_RECIPIENT_SUCCESS':
      return { ...state, loading: false, resultsSet: payload };

    case 'SEARCH_SUPPRESSIONS_RECIPIENT_FAIL': {
      return { ...state, loading: false };
    }

    default:
      return state;
  }
};
