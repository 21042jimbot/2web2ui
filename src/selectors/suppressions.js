import convertListToBoolHash from 'src/helpers/convertListToBoolHash';
import { createSelector } from 'reselect';

const selectSuppressionsSearch = (state) => state.suppressions.search;

const selectTypes = createSelector(
  [selectSuppressionsSearch],
  ({ types = []}) => types
);

const selectSources = createSelector(
  [selectSuppressionsSearch],
  ({ sources = []}) => sources
);

export const selectSearchInitialValues = createSelector(
  [selectTypes, selectSources],
  (types, sources) => ({
    types: convertListToBoolHash(types),
    sources: convertListToBoolHash(sources)
  })
);


