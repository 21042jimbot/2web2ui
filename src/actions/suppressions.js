/* eslint-disable max-lines */
import moment from 'moment';
import config from 'src/config';
import _ from 'lodash';
import csvFileParseRequest, { hasData, hasField } from 'src/actions/helpers/csvFileParseRequest';
import sparkpostApiRequest from 'src/actions/helpers/sparkpostApiRequest';
import setSubaccountHeader from 'src/actions/helpers/setSubaccountHeader';
import { showAlert } from './globalAlert';

const { apiDateFormat } = config;

export function checkSuppression() { //used in DashBoardPage to check if account has suppression
  const params = { sources: 'Manually Added', limit: 1 };

  return (dispatch, getState) => dispatch(
    sparkpostApiRequest({
      type: 'CHECK_SUPPRESSIONS',
      meta: {
        method: 'GET',
        url: '/suppression-list',
        params
      }
    })
  );
}

export function searchRecipient({ email, subaccountId } = {}) {
  const headers = {};
  if (subaccountId) {
    headers['x-msys-subaccount'] = subaccountId;
  }

  return (dispatch, getState) => dispatch(
    sparkpostApiRequest({
      type: 'SEARCH_SUPPRESSIONS_RECIPIENT',
      meta: {
        method: 'GET',
        url: `/suppression-list/${email}`,
        headers
      }
    }))
    .catch((err) => {
      const { response = {}} = err;
      if (response.status === 404) {
        return; //swallow 404s
      }

      dispatch(
        showAlert({
          type: 'error',
          message: 'Error while searching recipient in suppressions list',
          details: err.message
        })
      );
    });
}

export function searchSuppressions(options) {
  const { dateOptions, types = [], sources = []} = options;
  const { from, to } = dateOptions;

  const params = {};

  if (from) {
    params.from = moment(from).utc().format(apiDateFormat);
  }

  if (to) {
    params.to = moment(to).utc().format(apiDateFormat);
  }

  if (!_.isEmpty(types)) {
    params.types = types.join(',');
  }

  if (!_.isEmpty(sources)) {
    params.sources = sources.join(',');
  }

  return (dispatch) => dispatch(
    sparkpostApiRequest({
      type: 'GET_SUPPRESSIONS',
      meta: {
        method: 'GET',
        url: '/suppression-list',
        params
      }
    }))
    .catch((err) => {
      dispatch(
        showAlert({
          type: 'error',
          message: 'Error while searching suppressions list',
          details: err.message
        })
      );
    });
}

export function deleteSuppression(suppression) {
  const { recipient, subaccount_id: subaccountId, type } = suppression;

  return sparkpostApiRequest({
    type: 'DELETE_SUPPRESSION',
    meta: {
      method: 'DELETE',
      url: `/suppression-list/${recipient}`,
      headers: setSubaccountHeader(subaccountId),
      data: { type },
      suppression
    }
  });
}

const LIKE_NON = new RegExp('non', 'i');
const LIKE_TRUE = new RegExp('true', 'i');

// SEE: https://developers.sparkpost.com/api/suppression-list.html#suppression-list-bulk-insert-update-put
export function createOrUpdateSuppressions(recipients, subaccount) {
  const sanitizedRecipients = recipients.map(({
    description, email, non_transactional, recipient, transactional, type
  }) => {
    // Convert deprecated type fields
    if (!type && LIKE_TRUE.test(transactional)) { type = 'transactional'; }
    if (!type && LIKE_TRUE.test(non_transactional)) { type = 'non_transactional'; }

    // Format type value to provide a better user experience
    if (type && LIKE_NON.test(type)) { type = 'non_transactional'; }
    if (type && !LIKE_NON.test(type)) { type = 'transactional'; }

    // Convert deprecated recipient fields
    if (email && !recipient) { recipient = email; }

    // Trim whitespace from recipient email (FAD-5095)
    return { description, recipient: _.trim(recipient), type };
  });

  return sparkpostApiRequest({
    type: 'CREATE_OR_UPDATE_SUPPRESSIONS',
    meta: {
      method: 'PUT',
      url: '/suppression-list',
      headers: setSubaccountHeader(subaccount),
      data: {
        recipients: sanitizedRecipients
      }
    }
  });
}

export function parseSuppressionsFile(file) {
  return csvFileParseRequest({
    type: 'PARSE_SUPPRESSIONS_FILE',
    meta: {
      file,
      validate: [
        hasData,
        hasField('recipient', 'email'),
        hasField('type', 'non_transactional', 'transactional')
      ]
    }
  });
}

export function uploadSuppressions(file, subaccount) {
  return async(dispatch) => {
    const recipients = await dispatch(parseSuppressionsFile(file));
    return dispatch(createOrUpdateSuppressions(recipients, subaccount));
  };
}

export function resetErrors() {
  return {
    type: 'RESET_SUPPRESSION_ERRORS'
  };
}

/**
 * Refreshes the date range for message events
 *
 * Calculates relative ranges if a non-custom relativeRange value is present,
 * which will override passed in from/to dates
 *
 * @param {Object} dateOptions
 * @param {Date} dateOptions.from
 * @param {Date} dateOptions.to
 * @param {String} dateOptions.relativeRange
 */
export function refreshSuppressionDateRange(dateOptions) {
  return {
    type: 'REFRESH_SUPPRESSION_SEARCH_DATE_OPTIONS',
    payload: dateOptions
  };
}

export function updateSuppressionSearchTypes(types) {
  return {
    type: 'UPDATE_SUPPRESSION_SEARCH_TYPES',
    payload: types
  };
}

export function updateSuppressionSearchSources(sources) {
  return {
    type: 'UPDATE_SUPPRESSION_SEARCH_SOURCES',
    payload: sources
  };
}
