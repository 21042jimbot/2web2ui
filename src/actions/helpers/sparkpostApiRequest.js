/**
 * Creates "actions" for a SparkPost API request, dispatching
 * the 3 necessary actions for each request.
 */

import _ from 'lodash';
import { refresh, logout } from 'actions/auth';
import { received } from 'actions/apiFailure';
import { sparkpostRequest, useRefreshToken } from 'helpers/http';
import { resolveOnCondition } from 'helpers/promise';

const maxRefreshRetries = 3;
let refreshing = false;
const refreshTokensUsed = new Set();

// Re-dispatches a given action after we finish refreshing the auth token
function redispatchAfterRefresh(action) {
  return resolveOnCondition(() => !refreshing).then(() => sparkpostApiRequest(action));
}

export default function sparkpostApiRequest(action) {
  return (dispatch, getState) => {
    const { auth } = getState();
    const { type = 'NO_TYPE_DEFINED', meta } = action;
    const { url, method = 'get', params, headers, data, onSuccess, retries = 0 } = meta;
    const PENDING_TYPE = `${type}_PENDING`;
    const SUCCESS_TYPE = `${type}_SUCCESS`;
    const FAIL_TYPE = `${type}_FAIL`;

    dispatch({
      type: PENDING_TYPE,
      meta
    });

    const httpOptions = {
      method: method.toLowerCase(),
      url,
      params,
      headers,
      data
    };

    if (auth.loggedIn) {
      _.set(httpOptions, 'headers.Authorization', auth.token);
    }

    // if we are currently refreshing our token, wait for refresh
    // to complete and then re-dispatch the action so it uses new token
    if (refreshing) {
      return redispatchAfterRefresh(action);
    }

    return sparkpostRequest(httpOptions)
      .then(({ data: { results }}) => {
        // we only get here if the request returned a 2xx status code
        dispatch({
          type: SUCCESS_TYPE,
          payload: results,
          meta
        });

        if (typeof onSuccess === 'function') {
          dispatch({
            type: 'OOPS_ON_SUCCESS_FUNC',
            payload: 'You shouldn\'t be using onSuccess callbacks anymore.'
          });
        }
      },
      // API request failed
      ({ message, response = {}}) => {
        // NOTE: if this is a 401 and we have a refresh token, we need to do a
        // refresh to get a new auth token and then re-dispatch this action
        if (response.status === 401 && auth.refreshToken && retries <= maxRefreshRetries) {
          action.meta.retries = retries + 1;

          // If we are currently refreshing the token OR if this refresh token
          // has already been used to refresh, we should re-dispatch after refresh is complete
          if (refreshing || refreshTokensUsed.has(auth.refreshToken)) {
            return redispatchAfterRefresh(action);
          }

          refreshing = true;
          refreshTokensUsed.add(auth.refreshToken);

          // call API for a new token
          return useRefreshToken(auth.refreshToken)

            // dispatch a refresh action to save new token results in cookie and store
            .then(({ data } = {}) => dispatch(refresh(data.access_token, data.refresh_token)))

            // dispatch the original action again, now that we have a new token ...
            // if anything in this refresh flow blew up, log out
            .then(
              // refresh token request succeeded
              () => {
                refreshing = false;
                return sparkpostApiRequest(action);
              },
              // refresh token request failed
              () => dispatch(logout())
            );
        }

        // If we have a 403 or a 401 and we're not refreshing, log the user out silently
        if (response.status === 401 || response.status === 403) {
          return dispatch(logout());
        }

        // any other API error should automatically fail, to be handled in the reducers/components
        dispatch({
          type: FAIL_TYPE,
          payload: { message, response },
          meta
        });

        if (response.status >= 500) {
          dispatch(received({ message, response }, meta));
        }
      });
  };
}
