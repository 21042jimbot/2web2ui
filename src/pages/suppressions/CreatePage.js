import _ from 'lodash';
import React, { Component } from 'react';
import { Link, withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import { Field, reduxForm, SubmissionError } from 'redux-form';
import { Button, Page, Panel } from '@sparkpost/matchbox';

import { showAlert } from 'src/actions/globalAlert';
import { uploadSuppressions } from 'src/actions/suppressions';
import ApiErrorBanner from 'src/components/apiErrorBanner/ApiErrorBanner';
import DownloadLink from 'src/components/downloadLink/DownloadLink';
import FileFieldWrapper from 'src/components/reduxFormWrappers/FileFieldWrapper';
import SubaccountTypeaheadWrapper from 'src/components/reduxFormWrappers/SubaccountTypeaheadWrapper';
import config from 'src/config';
import { fileExtension, fileWithContents, maxFileSize, required } from 'src/helpers/validation';
import exampleSuppressionsListPath from './example-suppressions-list.csv';

export class CreatePage extends Component {

  // Must re-throw Error as SubmissionError for redux form
  handleSubmit = ({ subaccount, suppressionsFile }) => (
    this.props.uploadSuppressions(suppressionsFile, subaccount)
      .then(this.handleSubmitSuccess)
      .catch((error) => { throw new SubmissionError(error); })
  )

  handleSubmitSuccess = () => {
    const { history, showAlert } = this.props;

    showAlert({ message: 'Successfully updated your suppression list', type: 'success' });
    history.push('/lists/suppressions');
  }

  get errorDetails() {
    const { parseError, persistError } = this.props;

    return parseError
      ? parseError.details || parseError.message
      : _.get(persistError, 'response.data.errors[0].message', persistError.message);
  }

  render() {
    const { handleSubmit: reduxFormSubmit, submitFailed, submitting } = this.props;

    return (
      <Page
        title="Add Suppressions"
        breadcrumbAction={{ content: 'Suppressions', Component: Link, to: '/lists/suppressions' }}
      >
        <form onSubmit={reduxFormSubmit(this.handleSubmit)}>
          {!submitting && submitFailed && (
            <ApiErrorBanner
              errorDetails={this.errorDetails}
              message="Unable to upload your list of suppressions"
              status="danger"
            />
          )}
          <Panel title="Upload a List of Suppressions">
            <Panel.Section>
              <Field
                component={FileFieldWrapper}
                disabled={submitting}
                name="suppressionsFile"
                label="Comma Separated Values (CSV) File of Suppressions"
                fileType="csv"
                helpText={
                  <span>
                    You can download
                    an <DownloadLink href={exampleSuppressionsListPath}>example file here</DownloadLink> to
                    use when formatting your list of suppressions for upload.
                  </span>
                }
                required
                validate={[
                  required,
                  fileExtension('csv'),
                  maxFileSize(config.maxUploadSizeBytes),
                  fileWithContents
                ]}
              />
              <Field
                component={SubaccountTypeaheadWrapper}
                disabled={submitting}
                helpText="Leaving this field blank will add the suppressions to the master account."
                name="subaccount"
              />
            </Panel.Section>
            <Panel.Section>
              <Button primary disabled={submitting} type="submit">Upload</Button>
            </Panel.Section>
          </Panel>
        </form>
      </Page>
    );
  }
}

const FORM_NAME = 'uploadSuppressions';
const mapStateToProps = (state) => ({
  persistError: state.suppressions.persistError,
  parseError: state.suppressions.parseError
});

export default withRouter(connect(mapStateToProps, { showAlert, uploadSuppressions })(reduxForm({ form: FORM_NAME })(CreatePage)));
