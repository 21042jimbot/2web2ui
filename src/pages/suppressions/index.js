import CreatePage from './CreatePage';
import ListPage from './ListPage';

export default {
  CreatePage,
  ListPage
};
