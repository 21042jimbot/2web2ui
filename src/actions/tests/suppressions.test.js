import sparkpostApiRequest from 'src/actions/helpers/sparkpostApiRequest';
import * as suppressions from '../suppressions';
import csvFileParseRequest, { hasField } from 'src/actions/helpers/csvFileParseRequest';

jest.mock('src/actions/helpers/sparkpostApiRequest', () => jest.fn((a) => a));
jest.mock('src/actions/helpers/csvFileParseRequest');

// Have to manually mock to avoid overrite of named exports
csvFileParseRequest.mockImplementation((a) => a);
hasField.mockImplementation((a) => jest.fn(a));

describe('Action Creator: Suppressions', () => {
  describe('deleteSuppression', () => {
    let mockSuppression;

    beforeEach(() => {
      mockSuppression = { recipient: 'foo/@bar.com', type: 'non_transactional' } ;
    });

    it('makes api call with correct parameter', async () => {
      await suppressions.deleteSuppression(mockSuppression);
      expect(sparkpostApiRequest).toHaveBeenCalledTimes(1);
      expect(sparkpostApiRequest).toHaveBeenCalledWith(
        {
          type: 'DELETE_SUPPRESSION',
          meta: {
            method: 'DELETE',
            url: '/v1/suppression-list/foo%2F%40bar.com',
            headers: {},
            data: { type: mockSuppression.type },
            suppression: mockSuppression
          }
        }
      );
    });

    it('includes subaccount id in header when exists', async () => {
      mockSuppression.subaccount_id = 101;
      await suppressions.deleteSuppression(mockSuppression);
      expect(sparkpostApiRequest).toHaveBeenCalledTimes(1);
      expect(sparkpostApiRequest).toHaveBeenCalledWith(
        {
          type: 'DELETE_SUPPRESSION',
          meta: {
            method: 'DELETE',
            url: '/v1/suppression-list/foo%2F%40bar.com',
            headers: { 'x-msys-subaccount': 101 },
            data: { type: mockSuppression.type },
            suppression: mockSuppression
          }
        }
      );
    });
  });

  describe('createOrUpdateSuppressions', () => {
    let recipients;

    beforeEach(() => {
      recipients = [{ recipient: ' example@test.com ', type: 'transactional' }];
    });

    it('without a subaccount', () => {
      const action = suppressions.createOrUpdateSuppressions(recipients);
      expect(action).toMatchSnapshot();
    });

    it('with a subaccount', () => {
      const subaccount = 999;
      const action = suppressions.createOrUpdateSuppressions(recipients, subaccount);

      expect(action).toMatchSnapshot();
    });

    it('transforms deprecated email to recipient field', () => {
      const action = suppressions.createOrUpdateSuppressions([
        { email: 'example@test.com', type: 'transactional' }
      ]);

      expect(action).toMatchSnapshot();
    });

    it('ignores deprecated email field when recipient field is present', () => {
      const action = suppressions.createOrUpdateSuppressions([
        { email: 'old@test.com', recipient: 'example@test.com', type: 'transactional' }
      ]);

      expect(action).toMatchSnapshot();
    });

    it('transforms deprecated non_transactional and transactional fields to type', () => {
      const action = suppressions.createOrUpdateSuppressions([
        { recipient: 'example@test.com', non_transactional: 'true', transactional: 'false' },
        { recipient: 'example@test.com', non_transactional: 'false', transactional: 'true' }
      ]);

      expect(action).toMatchSnapshot();
    });

    it('ignores non_transactional and transactional fields when type is present', () => {
      const action = suppressions.createOrUpdateSuppressions([
        {
          recipient: 'transactional@test.com',
          non_transactional: 'true',
          transactional: 'false',
          type: 'transactional'
        },
        {
          recipient: 'non_transactional@test.com',
          non_transactional: 'false',
          transactional: 'true',
          type: 'non_transactional'
        }
      ]);

      expect(action).toMatchSnapshot();
    });

    it('transform non-standard type field values', () => {
      const action = suppressions.createOrUpdateSuppressions([
        { recipient: 'example@test.com', type: 'non_marketing' },
        { recipient: 'example@test.com', type: 'marketing' }
      ]);

      expect(action).toMatchSnapshot();
    });
  });

  describe('parseSuppressionsFile', () => {
    it('request to parse local file', () => {
      const localFile = { name: 'example.csv', size: 123 };
      const action = suppressions.parseSuppressionsFile(localFile);

      expect(hasField).toHaveBeenCalledWith('recipient', 'email');
      expect(hasField).toHaveBeenCalledWith('type', 'non_transactional', 'transactional');
      expect(action).toMatchSnapshot();
    });
  });

  describe('uploadSuppressions', () => {
    it('parses local file and creates suppressions', async () => {
      const localFile = { name: 'example.csv', size: 123 };
      const recipients = [{ recipient: ' example@test.com ', type: 'transactional' }];
      const subaccount = 999;
      const mockDispatch = jest.fn(() => Promise.resolve())
        .mockImplementationOnce(() => Promise.resolve(recipients));
      const thunk = suppressions.uploadSuppressions(localFile, subaccount);

      await thunk(mockDispatch);

      expect(mockDispatch.mock.calls).toMatchSnapshot();
    });
  });
});
