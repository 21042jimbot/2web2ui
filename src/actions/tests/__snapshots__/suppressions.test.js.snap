// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Action Creator: Suppressions createOrUpdateSuppressions ignores deprecated email field when recipient field is present 1`] = `
Object {
  "meta": Object {
    "data": Object {
      "recipients": Array [
        Object {
          "description": undefined,
          "recipient": "example@test.com",
          "type": "transactional",
        },
      ],
    },
    "headers": Object {},
    "method": "PUT",
    "url": "/suppression-list",
  },
  "type": "CREATE_OR_UPDATE_SUPPRESSIONS",
}
`;

exports[`Action Creator: Suppressions createOrUpdateSuppressions transform non-standard type field values 1`] = `
Object {
  "meta": Object {
    "data": Object {
      "recipients": Array [
        Object {
          "description": undefined,
          "recipient": "example@test.com",
          "type": "non_transactional",
        },
        Object {
          "description": undefined,
          "recipient": "example@test.com",
          "type": "transactional",
        },
      ],
    },
    "headers": Object {},
    "method": "PUT",
    "url": "/suppression-list",
  },
  "type": "CREATE_OR_UPDATE_SUPPRESSIONS",
}
`;

exports[`Action Creator: Suppressions createOrUpdateSuppressions transforms deprecated email to recipient field 1`] = `
Object {
  "meta": Object {
    "data": Object {
      "recipients": Array [
        Object {
          "description": undefined,
          "recipient": "example@test.com",
          "type": "transactional",
        },
      ],
    },
    "headers": Object {},
    "method": "PUT",
    "url": "/suppression-list",
  },
  "type": "CREATE_OR_UPDATE_SUPPRESSIONS",
}
`;

exports[`Action Creator: Suppressions createOrUpdateSuppressions with a subaccount 1`] = `
Object {
  "meta": Object {
    "data": Object {
      "recipients": Array [
        Object {
          "description": undefined,
          "recipient": "example@test.com",
          "type": "transactional",
        },
      ],
    },
    "headers": Object {
      "x-msys-subaccount": 999,
    },
    "method": "PUT",
    "url": "/suppression-list",
  },
  "type": "CREATE_OR_UPDATE_SUPPRESSIONS",
}
`;

exports[`Action Creator: Suppressions createOrUpdateSuppressions with deprecated non_transactional and transactional field 1`] = `
Object {
  "meta": Object {
    "data": Object {
      "recipients": Array [
        Object {
          "description": undefined,
          "recipient": "example@test.com",
          "type": "non_transactional",
        },
        Object {
          "description": undefined,
          "recipient": "example@test.com",
          "type": "transactional",
        },
      ],
    },
    "headers": Object {},
    "method": "PUT",
    "url": "/suppression-list",
  },
  "type": "CREATE_OR_UPDATE_SUPPRESSIONS",
}
`;

exports[`Action Creator: Suppressions createOrUpdateSuppressions without a subaccount 1`] = `
Object {
  "meta": Object {
    "data": Object {
      "recipients": Array [
        Object {
          "description": undefined,
          "recipient": "example@test.com",
          "type": "transactional",
        },
      ],
    },
    "headers": Object {},
    "method": "PUT",
    "url": "/suppression-list",
  },
  "type": "CREATE_OR_UPDATE_SUPPRESSIONS",
}
`;

exports[`Action Creator: Suppressions parseSuppressionsFile request to parse local file 1`] = `
Object {
  "meta": Object {
    "file": Object {
      "name": "example.csv",
      "size": 123,
    },
    "validate": Array [
      [MockFunction],
      [MockFunction],
      [MockFunction],
    ],
  },
  "type": "PARSE_SUPPRESSIONS_FILE",
}
`;

exports[`Action Creator: Suppressions uploadSuppressions parses local file and creates suppressions 1`] = `
Array [
  Array [
    Object {
      "meta": Object {
        "file": Object {
          "name": "example.csv",
          "size": 123,
        },
        "validate": Array [
          [MockFunction],
          [MockFunction],
          [MockFunction],
        ],
      },
      "type": "PARSE_SUPPRESSIONS_FILE",
    },
  ],
  Array [
    Object {
      "meta": Object {
        "data": Object {
          "recipients": Array [
            Object {
              "description": undefined,
              "recipient": "example@test.com",
              "type": "transactional",
            },
          ],
        },
        "headers": Object {
          "x-msys-subaccount": 999,
        },
        "method": "PUT",
        "url": "/suppression-list",
      },
      "type": "CREATE_OR_UPDATE_SUPPRESSIONS",
    },
  ],
]
`;
